### DISCUSSION TITLE
EFCore 2.2 blocks running on .NET 6

### DISCUSSION THREAD

**Discussion by huoyaoyuan** [ppy/osu#15544] [Ideas] [answered]
When trying to run osu!lazer on .NET 6, the static constructor of some EFCore type throws. (The type initializer for `Microsoft.EntityFrameworkCore.Query.ResultOperators.Internal.TrackingExpressionNode` threw an exception)
The inner exception is:
`InvalidOperationException: Sequence contains more than one matching element`

As a .NET runtime contributor (yeah I'm now a community member of them), I know that we added many handful overload methods in .NET 6. The EFCore infrastructure should be doing reflection to find some method, and it expected there's only 1 overload, which is broken by newly added overload.

Effectively, EFCore 2.2 is preventing osu! to get any benefit from .NET 6. We should consider to upgrade or move away from it to unblock further upgrades.

**Comment by bdach** [comment, marked as answer]
which is why we're actively in the middle of migrating away from EF to realm.

**Comment by smoogipoo** [comment]
Fyi, I've been maintaining a 6.0 branch on my fork: https://github.com/smoogipoo/osu/tree/net6.0

**Reply by huoyaoyuan** [reply]
Have you tried to migrate from Xamarin to `net6.0-ios` and `net6.0-android`?

**Reply by smoogipoo** [reply]
Nope
